AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthenticationProviderTests/AuthenticationProviderSigninTests.swift

```diff
@@ -39,7 +50,716 @@ class AuthenticationProviderSigninTests: BaseAuthenticationProviderTest {
+    ///
+    func testSignInWithInvalidLambdaResponseException() {
+        let error = AWSMobileClientError.invalidLambdaResponse(message: "Error")
+        mockAWSMobileClient.signInMockResult = .failure(error)
+
+        let options = AuthSignInRequest.Options()
+        let resultExpectation = expectation(description: "Should receive a result")
+        _ = plugin.signIn(username: "username", password: "password", options: options) { result in
+            defer {
+                resultExpectation.fulfill()
+            }
+            switch result {
+            case .success(let signinResult):
+                XCTFail("Should not produce result - \(signinResult)")
+            case .failure(let error):
+                guard case .service(_, _, let underlyingError) = error,
+                      case .lambda = (underlyingError as? AWSCognitoAuthError) else {
+                    XCTFail("Should produce lambda error but instead produced \(error)")
+                    return
+                }
+            }
+        }
+        wait(for: [resultExpectation], timeout: apiTimeout)
+    }
+
+    /// Test a signIn with `InvalidParameterException` from service
+    ///
+    /// - Given: Given an auth plugin with mocked service. Mocked service should mock a
+    ///   InvalidParameterException response for signIn
+    ///
+    /// - When:
+    ///    - I invoke signIn
+    /// - Then:
+    ///    - I should get a .service error with .invalidParameter error
+    ///
+    func testSignInWithInvalidParameterException() {
+        let error = AWSMobileClientError.invalidParameter(message: "Error")
+        mockAWSMobileClient.signInMockResult = .failure(error)
+
+        let options = AuthSignInRequest.Options()
+        let resultExpectation = expectation(description: "Should receive a result")
+        _ = plugin.signIn(username: "username", password: "password", options: options) { result in
+            defer {
+                resultExpectation.fulfill()
+            }
+            switch result {
+            case .success(let signinResult):
+                XCTFail("Should not produce result - \(signinResult)")
+            case .failure(let error):
+                guard case .service(_, _, let underlyingError) = error,
+                      case .invalidParameter = (underlyingError as? AWSCognitoAuthError) else {
+                    XCTFail("Should produce invalidParameter error but instead produced \(error)")
+                    return
+                }
+            }
+        }
+        wait(for: [resultExpectation], timeout: apiTimeout)
+    }
+
+    /// Test a signIn with `InvalidUserPoolConfigurationException` from service
+    ///
+    /// - Given: Given an auth plugin with mocked service. Mocked service should mock a
+    ///   InvalidUserPoolConfigurationException response for signIn
+    ///
+    /// - When:
+    ///    - I invoke signIn
+    /// - Then:
+    ///    - I should get a .configuration
```

Comment:
nit: I should get a .configuration error?

Reply:
Fixed